xbmc/games/controllers/Controller.cpp

```diff
@@ -122,16 +122,17 @@ bool CController::LoadLayout(void)
 
     CLog::Log(LOGINFO, "Loading controller layout: {}", CURL::GetRedacted(strLayoutXmlPath));
 
-    CXBMCTinyXML xmlDoc;
+    CXBMCTinyXML2 xmlDoc;
     if (!xmlDoc.LoadFile(strLayoutXmlPath))
     {
-      CLog::Log(LOGDEBUG, "Unable to load file: {} at line {}", xmlDoc.ErrorDesc(),
-                xmlDoc.ErrorRow());
+      CLog::Log(LOGDEBUG, "Unable to load file: {} at line {}", xmlDoc.ErrorStr(),
+                xmlDoc.ErrorLineNum());
       return false;
     }
 
-    TiXmlElement* pRootElement = xmlDoc.RootElement();
-    if (!pRootElement || pRootElement->NoChildren() || pRootElement->ValueStr() != LAYOUT_XML_ROOT)
+    auto* pRootElement = xmlDoc.RootElement();
+    if (!pRootElement || pRootElement->NoChildren() ||
+        strcmp(pRootElement->Value(), LAYOUT_XML_ROOT) != 0)
```

Comment:
Is it not possible to have c++ style string comparisons anymore?

Reply:
Tinyxml2 no longer uses strong in any way. So the question becomes, what's faster, string creation and then empty check or strcmp?

I'm easy either way if someone has their preference

Reply:
`strcmp` is going to be faster (but not noticeably), I just hate the inversion of the logic when `!= 0` means `equals`. It opens up so many possibilities for mistakes and decreases readability.

But thus is what the C++ gods hath giveth ye.

(And please use `std::strcmp and include the C++ header)

Reply:
Maybe just add a StringUtils function that wraps `std::strcmp() != 0` purely for readability.